### DISCUSSION TITLE
a new Untitled-1 file is opened everytime copilot fetches new information

### DISCUSSION THREAD

**Discussion by Mariownyou** [community/community#73722] [Copilot Conversations]
### Select Topic Area

Bug

### Body

How to reproduce the issue:

Load any vscode project
Edit any file that has copilot enabled
Create a new Untitled-1 file and exit the file without saving
Go to the original file and make any change, a Untitled-1 file is created
Close Untitled-1 and repeat 4

**Comment by svivian** [comment]
Here is an earlier discussion, which inexplicably was closed as a duplicate on this one: https://github.com/orgs/community/discussions/45269

Does this issue not infuriate all Github employees daily? Or do none of you actually use your own software?

**Reply by Mariownyou** [reply]
Agreed this is very annoying

**Reply by svipas** [reply]
Yeah and this is happening for a very long time...

**Comment by svivian** [comment]
It's a whole year since the original issue was reported and this still happens daily!! Surely the few of use can't be the only ones experiencing this issue?!

**Comment by P403n1x87** [comment]
I've been using GH Copilot for quite some time now, but this issue only appeared recently for me. It might have coincided with my personal account now getting the GH Copilot settings from the organisation I belong to (recently the chat feature has been enabled by my org).

**Comment by mark-turner** [comment]
Adding a me too here. The issue has been going on ever since Copilot was released. Bizarre that it hasn't been resolved.

**Comment by svivian** [comment]
Just a reminder, this is still a daily occurrence. Any time I make a new file (Cmd+N) then either save or close it, Copilot creates new blank files randomly every few minutes. I have to quit VSCode entirely and reopen to get it to stop.

Do any Github/Copilot developers read these issues? Can you comment on the solution given [in the previous thread](https://github.com/orgs/community/discussions/45269), is that viable or are there complications? Any acknowledgement would be a start. The extension is seemingly updated regularly so you must be doing *something*.

**Reply by rboadalabelgrup** [reply]
I am using the latest version of vscode and copilot and this error still happens and is very annoying.

**Comment by rboadalabelgrup** [comment]
I get daily updates of the copilot extension but this bug is still not fixed.

**Comment by benzdouglas** [comment]
Has there been any fix or workaround? Every time I ask Copilot something I get three tabs - 'Untitled-1', 'Untitled-2', and 'Untitled-3'. I tried the other solutions in the thread without any changes. It's extremely annoying to deal with this... Especially for a product that I pay for.

**Reply by j4k0xb** [reply]
I made a (manual) workaround a while ago that still works for the latest version: https://github.com/orgs/community/discussions/45269#discussioncomment-6223942

Would probably be a good idea to make an extension to automatically patch copilot on update

**Reply by benzdouglas** [reply]
Thanks, but I tried that workaround on 1.246.0 earlier - first tried with a manual find and replace which removed one of the three Untitled tabs generating for me, but then it returned somehow. Just tried the sed script to replace it and it didn't work :/

There were some other strange references to "untitled" in the extension.js, like:
`getRelativePath(t){if(!t.uri.startsWith("untitled:")`
and 
`detectLanguage(n){return n.uri.startsWith("untitled:")`
I'm not sure if that has anything to do with it...

edit: Also just tried to replace any instance of status==="valid" to include the check for the untitled uri scheme and that didn't do anything either

**Reply by j4k0xb** [reply]
> but then it returned somehow.

copilot frequently auto-updates which overwrites `extension.js` again.
you also need to restart vscode after editing it.

> There were some other strange references to "untitled" in the extension.js
> I'm not sure if that has anything to do with it...

Hard to say... I originally found it with a debugger. Look at the comment edit history of August 10, 2023 if you wanna know the involved functions and stack trace that lead to creating the untitled tab
And https://stackoverflow.com/questions/51164279/how-to-debug-visual-studio-code-extensions

**Reply by benzdouglas** [reply]
That original code is helpful! I'll look into it later today and see if I can find anything and report back. Thanks!

**Reply by Acme2020** [reply]
Same here. None of the described workarounds fixed it so far.

**Reply by j4k0xb** [reply]
@benzdouglas did you find anything?

**Reply by benzdouglas** [reply]
@j4k0xb Sorry about that, I needed to do Thanksgiving stuff and then just completely forgot when returning, and meant to put some time aside in the past few days to look at it but was too busy. But just now I noticed that Copilot had an update (Copilot 1.251.0 and Copilot Chat 0.23.1) and now it seems that the Untitled documents are no longer generating! I also had checked my laptop a few days ago and it wasn't generating there, so my best guess is it was caused by some kind of document ignoring setting I had locally to my desktop computer. Sorry for not following up sooner!

**Comment by ARBhosale** [comment]
Following command palette command resolved this issue for me.

`> Chat: Clear All Workspace Chats`

GitHub Copilot `v1.250.1260`
GitHub Copilot Chat `v0.23.2024102903`

**Comment by skyslide22** [comment]
still happening in 2025

**Comment by Ste1io** [comment]
Three years later, still a major problem on latest (un)stable build. And easy af to repro.

- Open a workspace (it must be a workspace)
- Now open a new untitled text window (`CTRL`+`N`)
- Now paste some text (JSON works) to trigger a language mode change (or set the language mode yourself)
- Now close the untitled window, selecting "No" in the save prompt

Even better, open `C:\Users\vibecoder\AppData\Roaming\Code\Backups\{workspaceIdentifier.id}\untited` and watch the fun happen in windows exploder while you play wack-a-tab.

(Direct link to that directory is in `C:\Users\vibecoder\AppData\Roaming\Code\User\globalStorage\storage.json` listed for each open vscode instance).

**Comment by chaitanya-mhetre** [comment]
This issue appears to be a long-standing interaction bug between VS Code’s backup/restore mechanism and Copilot Chat’s document context handling, rather than a simple UI glitch.

What seems to be happening (based on reports & repro steps)

When an Untitled editor:

changes language mode (explicitly or implicitly, e.g. pasting JSON), and

is then closed without saving inside a workspace,

VS Code still retains a backup entry for that document under the workspace backup directory.
Copilot Chat then appears to re-hydrate or request context from that orphaned backup during subsequent edits, which triggers VS Code to reopen it as a new Untitled-* tab.

The fact that users can observe this directly in:

%APPDATA%\Code\Backups\<workspaceId>\untitled


strongly suggests this is a backup lifecycle / cleanup issue, not just Copilot UI behavior.

Why “restart VS Code” or “Clear All Workspace Chats” helps

Both actions effectively:

Clear Copilot’s cached conversation/document context

Reset VS Code’s in-memory editor references

This aligns with reports that:

Chat: Clear All Workspace Chats


temporarily resolves the issue, while extension updates sometimes regress it.

Practical mitigations (until a fix lands)

These are not ideal, but they reduce impact:

Avoid language mode changes in untitled files
If testing snippets, paste them into a scratch file instead of Untitled.

Disable backup restore for untitled editors (advanced users)
While there’s no granular UI toggle, minimizing reliance on untitled buffers reduces repro frequency.

Clear workspace chats when it starts happening
Confirms the Copilot context link to the issue.

Reproducible minimal case
The steps described by @Ste1io are excellent and should be treated as the canonical repro.

Why this deserves attention

It’s easy to reproduce

It affects paid Copilot users

It creates editor noise that directly impacts productivity

It has existed across multiple Copilot versions and VS Code updates

At this point, an acknowledgement or tracked internal issue reference from the Copilot/VS Code team would already be a meaningful step forward.

Thanks to everyone who provided concrete repro steps and filesystem paths — that level of detail should make this fixable.
